libyul/optimiser/Suite.cpp

```diff
@@ -123,7 +127,10 @@ void OptimiserSuite::run(
 	// ForLoopInitRewriter. Run them first to be able to run arbitrary sequences safely.
 	suite.runSequence("hgfo", astRoot);
 
-	NameSimplifier::run(suite.m_context, astRoot);
+	{
+		PROFILER_PROBE("NameSimplifier", probe);
+		NameSimplifier::run(suite.m_context, astRoot);
+	}
```

Comment:
I wonder why we're running `NameSimplifier` twice. It has been that way ever since it was added in #9591. I don't see any specific justification for this in that PR though.

Removing it could shave off 2-3% of the optimizer running time.

Reply:
I was thinking that perhaps it yields nicer names in some cases, but at least for our test suite it [doesn't even change anything](https://app.circleci.com/pipelines/github/ethereum/solidity/36534/workflows/86d496ea-8695-4505-8527-e3e9ac4f117c).

Reply:
Nice. You could submit that as a PR then. We should run it by Daniel just to make sure there wasn't any obscure reason he's aware of but if it's having no effect, we should really get rid of it. It's too expensive.